src/test/java/uk/gov/hmcts/reform/civil/service/CoreCaseDataServiceTest.java

```diff
@@ -334,6 +337,61 @@ void shouldReturnGeneralApplications_WhenSearchingGeneralApplicationsAsSystemUpd
         }
     }
 
+    @Nested
+    class SearchGeneralApplicationWithCaseId {
+
+        @Test
+        void shouldReturnMoreThan10GeneralApplications_WhenSearchingGeneralApplicationsAsSystemUpdateUser() {
+            final Query query = new Query(matchQuery("data.generalAppParentCaseLink.CaseReference", CASE_ID),
+                                    List.of("data.applicationTypes",
+                                            "data.generalAppInformOtherParty.isWithNotice",
+                                            "data.generalAppRespondentAgreement.hasAgreed",
+                                            "data.parentClaimantIsApplicant",
+                                            "data.applicationIsUncloakedOnce",
+                                            "state",
+                                            "data.applicationIsCloaked",
+                                            "data.judicialDecisionRequestMoreInfo",
+                                            "data.generalAppPBADetails"),
+                                    0);
+
+            List<CaseDetails> cases = new ArrayList<>();
+            cases.add(CaseDetails.builder().id(1L).build());
+            cases.add(CaseDetails.builder().id(2L).build());
+            cases.add(CaseDetails.builder().id(3L).build());
+            cases.add(CaseDetails.builder().id(4L).build());
+            cases.add(CaseDetails.builder().id(5L).build());
+            cases.add(CaseDetails.builder().id(6L).build());
+            cases.add(CaseDetails.builder().id(7L).build());
+            cases.add(CaseDetails.builder().id(8L).build());
+            cases.add(CaseDetails.builder().id(9L).build());
+            cases.add(CaseDetails.builder().id(10L).build());
+            cases.add(CaseDetails.builder().id(11L).build());
+
+            SearchResult searchResult = SearchResult.builder()
+                .total(11)
+                .cases(cases).build();
+
+            when(coreCaseDataApi.searchCases(
+                USER_AUTH_TOKEN,
+                SERVICE_AUTH_TOKEN,
+                GENERAL_APPLICATION_CASE_TYPE,
+                query.toString()
+            ))
+                .thenReturn(searchResult);
```

Comment:
Could we adapt this test (or add another test) that exercises the loop condition at least once please?  Here the `while` will be false first time round so loop is exited immediately.

Reply:
fixed